src/axom/inlet/examples/nested_structs.cpp

```diff
@@ -0,0 +1,276 @@
+    os << "   Translation operator:\n";
+    os << fmt::format("      with vector: {0}\n", op.translate);
+    break;
+  case Operator::Type::Rotate:
+    os << "   Rotation operator:\n";
+    os << fmt::format("      with axis: {0}\n", op.axis);
+    os << fmt::format("      with center: {0}\n", op.center);
+    break;
+  case Operator::Type::Slice:
+    os << "   Slice operator:\n";
+    os << fmt::format("      with x-coord: {0}\n", op.x);
+    os << fmt::format("      with y-coord: {0}\n", op.y);
+    os << fmt::format("      with z-coord: {0}\n", op.z);
+    break;
+  }
+  return os;
+}
+
+struct Geometry
+{
+  std::vector<Operator> operators;
+  std::string format;
+  std::string path;
+  enum class Units
+  {
+    Centimeters,
+    Meters
+  } units;
+  int start_dim;
+  static void defineSchema(inlet::Table& table)
+  {
+    table.addString("format");
+    table.addString("path");
+    // A string is used to represent the enumeration
+    table.addString("units");
+    table.addInt("start_dimensions");
+    // addGenericArray is used to represent std::vector<T> where
+    // T is any non-primitive type
+    auto& ops_table = table.addGenericArray("operators");
+    Operator::defineSchema(ops_table);
+  }
+};
+
+template <>
+struct FromInlet<Geometry>
+{
+  Geometry operator()(const inlet::Table& base)
+  {
+    Geometry result;
+    result.format = base["format"];
+    result.path = base["path"];
+    result.start_dim = base["start_dimensions"];
+
+    // FIXME: Remove when PR #429 is merged
+    auto ops = base["operators"].get<std::unordered_map<int, Operator>>();
+    for(const auto& ele : ops)
+    {
+      result.operators.push_back(ele.second);
+    }
+    return result;
+  }
+};
+
+std::ostream& operator<<(std::ostream& os, const Geometry& geom)
+{
+  os << fmt::format("Geometry in format: '{0}'\n", geom.format);
+  os << fmt::format("  with path: '{0}'\n", geom.path);
+  for(const auto& op : geom.operators)
+  {
+    os << op;
+  }
+  return os;
+}
+
+struct Shape
+{
+  std::string name;
+  enum class Material
+  {
+    Steel,
+    Wood,
+    Plastic
+  } material;
+  Geometry geom;
+  static void defineSchema(inlet::Table& table)
+  {
+    table.addString("name");
+    table.addString("material");
+    // addStruct is used for a single instance of a user-defined type
+    auto& geom_table = table.addStruct("geometry");
+    Geometry::defineSchema(geom_table);
+  }
+};
+
+std::ostream& operator<<(std::ostream& os, const Shape& shape)
+{
+  os << fmt::format("Shape: '{0}'\n", shape.name);
+  os << shape.geom;
+  return os;
+}
+
+template <>
+struct FromInlet<Shape>
+{
+  Shape operator()(const inlet::Table& base)
+  {
+    Shape result;
+    result.name = base["name"];
+    std::string material = base["material"];
+    if(material == "steel")
+    {
+      result.material = Shape::Material::Steel;
+    }
+    result.geom = base["geometry"].get<Geometry>();
+    return result;
+  }
+};
+
+const std::string input = R"(
```

Comment:
👍 for the multiline string literal!

Presumably, clang-format is happy with it?

Reply:
Surprisingly, yes